### ISSUE TITLE
False positive for reportOverlappingOverload with overloads for Literal[False] and Literal[True]

### ISSUE THREAD

**Issue by conradhaupt** [microsoft/pyright#10737]
**Describe the bug**

Overloads of a function which differentiate between `Literal[True]` and `Literal[False]` for a parameter results in a `reportOverlappingOverload` error. The error message is `Overload 1 for "my_func" overlaps overload 2 and returns an incompatible type`.

For example, I may have a flag parameter which changes the returned type, to optionally return additional values. In the example, `my_func` can return tuples with two or three values, controlled by `return_two`.

**Code or Screenshots**
Here is a minimum working example, which I tested on https://pyright-play.net.

```python
from typing import overload, Literal


@overload
def my_func( # <-- Overload 1 for "my_func" overlaps overload 2 and returns an incompatible type.
    required: int,
    return_two: Literal[True] = ...,
) -> tuple[int, int]: ...


@overload
def my_func(
    required: int,
    return_two: Literal[False] = ...,
) -> tuple[int, int, int]: ...


def my_func(
    required: int,
    return_two: bool = True,
) -> tuple[int, int] | tuple[int, int, int]:
    if return_two:
        return (required, required)
    else:
        return (required, required, required)

```

**VS Code extension or command-line**
I have tested this on https://pyright-play.net and with BasedPyRight. As it appears to be present with PyRight, I'm reporting it here.

**Comment by erictraut** [comment]
This isn't a false positive. The problem is that both of your overloads have a second parameter with a default argument. If you call the overload with a single argument, both of the overloads apply, but their return types are contradictory. To fix the overlap, you need to remove the `= ...` from one of the two overloads.
